qt/python/mantidqt/mantidqt/widgets/codeeditor/scriptcompatibility.py

```diff
@@ -37,16 +36,6 @@ def attr_called(attr, content):
     return bool(re.search(r'(^|\s)+' + attr + r'\(.*\)', content))
 
 
-def check_future_import(content):
-    """Return line number of __future__ import in content, if present"""
-    match = re.search(r'(import .*__future__|from __future__  *import)',
-                      content)
-    if match:
-        line_no = len(re.findall(r'(\r\n|\n|\r)', content[:match.end()])) + 1
-        return line_no
-    return False
-
-
 def mantid_algorithm_used_without_import(content):
     for attr in dir(simpleapi):
         if (not attr.startswith('_') and attr == attr.title()
```

Comment:
Do you know why the attr is checked against its title here? I believe it causes some issues where the prompt to add the necessary imports will not be triggered because the title is usually in lower case (apart from the first character). Here's the example, follow the same procedure for the file you proposed in the test but fill it for example with the following:

```
res1 = CreateSampleWorkspace()
result = ConvertUnits(InputWorkspace=res1, Target='Wavelength')
SaveAscii(result, Filename='out')
```

I believe it makes sense that a prompt should appear to add the necessary import for that file, but it doesn't happen because these algorithms have an attribute title that is not exactly the same as the algorithm name. 

A simple fix would be to check on lower cases only, but I am not sure if there was a different reason for this check to be there.

Reply:
Good find!! Looked in the git history and I think this was an oversight! They wanted to stop things like sys and os, which seem to be in the simpleapi, from being checked for as algorithms. ![Screenshot 2022-09-28 at 10 38 52](https://user-images.githubusercontent.com/55980573/192745912-1e38903a-002d-4945-9c8e-88fa7f5957d4.png)